Documentation/BasicOperators.md

```diff
@@ -46,41 +46,394 @@ code-style will be used.
+sendCompleted(lettersSink)  // prints "Completed"
+```
+
+The `combineLatestWith` operator works in the same way, but as an operator.
+
 ### Zipping
 
+The `zip` function combines values of two signals into pairs. The elements of any Nth pair are the Nth elements of the two input signals. That means the output signal will always wait for both input signals to send and output.
+
+```Swift
+let (numbersSignal, numbersSink) = Signal<Int, NoError>.pipe()
+let (lettersSignal, lettersSink) = Signal<String, NoError>.pipe()
+
+zip(numbersSignal, lettersSignal)
+    |> observe(next: { println($0) },
+          completed: { println("Completed") })
+
+sendNext(numbersSink, 0)    // nothing printed
+sendNext(numbersSink, 1)    // nothing printed
+sendNext(lettersSink, "A")  // prints (1, A)
+sendNext(numbersSink, 2)    // prints (2, A)
+sendCompleted(numbersSink)  // nothing printed
+sendNext(lettersSink, "B")  // prints (2, B)
+sendNext(lettersSink, "C")  // prints (2, C) & "Completed"
+
+```
+
+The `zipWith` operator works in the same way, but as an operator.
 
 ## Flattening producers
 
-### Concatenating
+The `flatten` operators transforms a `SignalProducer`-of-`SignalProducer`s into a single `SignalProducer`.
+There are multiple different semantics of the operation which can be chosen as a `FlattenStrategy`.
+
+To understand, why there are different strategies and how they compare to each other, take a look at this example and
+imagine the column offsets as time:
+
+```Swift
+let values = [
+// imagine column offset as time
+[ 1,    2,      3 ],
+   [ 4,      5,     6 ],
+         [ 7,     8 ],
+]
+
+let merge =
+[ 1, 4, 2, 7,5, 3,8,6 ]
+
+let concat = 
+[ 1,    2,      3,4,      5,     6,7,     8]
+
+let latest =
+[ 1, 4,    7,     8 ]
+```
+
+Note, how the values interleave and which values are even included in the resulting array.
+
+
 ### Merging
+
+The `.Merge` strategy works by immediately forwarding every events of the inner `SignalProducer`s to the outer `SignalProducer`.
+
+```Swift
+let (numbersSignal, numbersSink) = SignalProducer<AnyObject, NoError>.buffer(5)
+let (lettersSignal, lettersSink) = SignalProducer<AnyObject, NoError>.buffer(5)
+let (signal, sink) = SignalProducer<SignalProducer<AnyObject, NoError>, NoError>.buffer(5)
+
+signal
+    |> flatten(FlattenStrategy.Merge)
+    |> start(next: { println($0) })
+
+sendNext(sink, numbersSignal)
+sendNext(sink, lettersSignal)
+sendCompleted(sink)
+
+sendNext(numbersSink, 1)    // prints 1
+sendNext(lettersSink, "A")  // prints A
+sendNext(numbersSink, 2)    // prints 2
+sendNext(lettersSink, "B")  // prints B
+sendNext(numbersSink, 3)    // prints 3
+sendNext(lettersSink, "C")  // prints C
+```
+
+### Concatenating
+
+The `.Concat` strategy works by concatenating the `SignalProducer`s such that their values are sent in the order of the `SignalProducer`s themselves. 
+This implies, that values of a `SignalProducer` are only sent, when the preceding `SignalProducer` has completed
```

Comment:
I'd reword these sentences a bit:

The `.Concat` strategy is used to serialize work of the inner `SignalProducer`s. The first producer sent by the outer producer is started immediately. Each subsequent producer is not started until the preceding one has completed. Errors are immediately forwarded to the flattened producer.